lib/cn/vblock_builder.c

```diff
@@ -193,10 +188,27 @@ vbb_create(
     bld->pc = pc;
     bld->ds = cn_get_dataset(cn);
     bld->vgroup = vgroup;
-    bld->max_size = rp->vblock_size;
     bld->agegroup = HSE_MPOLICY_AGE_LEAF;
     bld->wbuf = wbuf;
 
+    mp = cn_get_dataset(cn);
+
+    for (int i = HSE_MCLASS_BASE; i < HSE_MCLASS_COUNT; i++) {
+        struct mpool_mclass_props props;
+
+        if (!mpool_mclass_is_configured(mp, i))
+            continue;
+
+        err = mpool_mclass_props_get(mp, i, &props);
+        if (err) {
+            vlb_free(wbuf, WBUF_LEN_MAX + sizeof(*bld));
+            return err;
+        }
+
+        if (props.mc_mblocksz > bld->max_size)
+            bld->max_size = props.mc_mblocksz;
+    }
```

Comment:
In theory we can get out of this loop w/ `bld->max_size == 0`.  Maybe change to:
```
for (...) {
    err = mpool_mclass_props_get(mp, i, &props);
    if (err)
        break;
}
if (err || blk->max_size == 0) {
    vlb_free(...);
    return err ? err : merr(UGH);
}
```